src/test/java/dev/blaauwendraad/masker/randomgen/RandomWhiteSpacePrettyPrinter.java

```diff
@@ -0,0 +1,104 @@
+package dev.blaauwendraad.masker.randomgen;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.PrettyPrinter;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+
+import java.io.IOException;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static dev.blaauwendraad.masker.json.util.AsciiCharacter.*;
+
+/**
+ * This class is DefaultPrettyPrinter with additional logic to add white spaces to JSON
+ */
+public class RandomWhiteSpacePrettyPrinter implements PrettyPrinter {
+    private final DefaultPrettyPrinter defaultPrettyPrinter = new DefaultPrettyPrinter();
+
+    private int numberOfWhiteSpacesToInject;
+
+    public RandomWhiteSpacePrettyPrinter(int numberOfWhiteSpacesToInject) {
+        this.numberOfWhiteSpacesToInject = numberOfWhiteSpacesToInject;
+    }
+
+    @Override
+    public void writeRootValueSeparator(JsonGenerator gen) throws IOException {
+        defaultPrettyPrinter.writeRootValueSeparator(gen);
+    }
+
+    @Override
+    public void writeStartObject(JsonGenerator gen) throws IOException {
+        defaultPrettyPrinter.writeStartObject(gen);
+        addRandomWhiteSpace(gen);
+    }
+
+    @Override
+    public void writeEndObject(JsonGenerator gen, int nrOfEntries) throws IOException {
+        defaultPrettyPrinter.writeEndObject(gen, nrOfEntries);
+        addRandomWhiteSpace(gen);
+    }
+
+    @Override
+    public void writeObjectEntrySeparator(JsonGenerator gen) throws IOException {
+        defaultPrettyPrinter.writeObjectEntrySeparator(gen);
+        addRandomWhiteSpace(gen);
+    }
+
+    @Override
+    public void writeObjectFieldValueSeparator(JsonGenerator gen) throws IOException {
+        defaultPrettyPrinter.writeObjectFieldValueSeparator(gen);
+        addRandomWhiteSpace(gen);
+    }
+
+    @Override
+    public void writeStartArray(JsonGenerator gen) throws IOException {
+        defaultPrettyPrinter.writeStartArray(gen);
+        addRandomWhiteSpace(gen);
+    }
+
+    @Override
+    public void writeEndArray(JsonGenerator gen, int nrOfValues) throws IOException {
+        defaultPrettyPrinter.writeEndArray(gen, nrOfValues);
+        addRandomWhiteSpace(gen);
+    }
+
+    @Override
+    public void writeArrayValueSeparator(JsonGenerator gen) throws IOException {
+        defaultPrettyPrinter.writeArrayValueSeparator(gen);
+        addRandomWhiteSpace(gen);
+    }
+
+    @Override
+    public void beforeArrayValues(JsonGenerator gen) throws IOException {
+        defaultPrettyPrinter.beforeArrayValues(gen);
+        addRandomWhiteSpace(gen);
+    }
+
+    @Override
+    public void beforeObjectEntries(JsonGenerator gen) throws IOException {
+        defaultPrettyPrinter.beforeObjectEntries(gen);
+        addRandomWhiteSpace(gen);
```

Comment:
Does it make sense to inject the whitespace not only after, but also before (here and in other places)?